husc/screens/cellomics.py

```diff
@@ -0,0 +1,366 @@
+"""Feature computations and other functions for Cellomics screens.
+"""
+
+import os
+import collections as coll
+from skimage import io
+import numpy as np
+from cytoolz import groupby
+
+
+def listdir_fullpath(path):
+    """Extended os.listdir that lists the full directory instead of just the
+    filename.
+
+    Parameters
+    ----------
+    dir : string
+        The query directory.
+
+    Returns
+    -------
+    path : list of string
+        A list of files in the query directory with their full path.
+    """
+    path = [os.path.join(path, fn) for fn in os.listdir(path)]
+    return path
+
+
+def batch_stitch_stack(file_dict, output, order=[2, 1, 0], bit=None):
+    """Run snail stitch and concatenate the channels across a set of images.
+
+    This function takes the (plate, well) dictionary built using the
+    ``make_key2file`` function. Images are grouped according to their channel,
+    stitched together and stacked into a single 3-channel image. Images
+    are re-scaled and saved to a user specified output directory. Images
+    are saved to directories according to their plate number.
+
+    Parameters
+    ----------
+    file_dict : dict { tuple (plate, well) : list of strings }
+        The dictionary mapping the (plate, well) tuple to a list of image
+        files. This dictionary is built using the ``make_key2file`` function.
+    output : string
+        The directory to output the stitched and concatenated images to.
+    order : list of int, optional
+        The order the channels should be in in the final image. Default
+        [0, 1, 2].
+    bit : string, optional
+        If images need to be rescaled, select whether or not to rescale the
+        images. Value should be 8 or 16 for 8 and 16 bit scaling respectively.
+    """
+    for fns in file_dict.values():
+        fn0 = fns[0]
+        sem = cellomics_semantic_filename(fns[0])
+        plate = str(sem['plate'])
+        fn0 = os.path.split(fn0)[1]
+        fn0_fn, fn_ext = os.path.splitext(fn0)
+        new_fn = [fn0_fn, '_stitched', fn_ext]
+        new_fn = ''.join(new_fn)
+
+        channels = groupby(channel, fns)
+        while len(channels) < 3:
+            channels[len(channels)] = None
+
+        images = []
+        for fns in channels.values():
+            if fns is None:
+                images.append(None)
+            else:
+                image = snail_stitch(fns)
+                images.append(image)
```

Comment:
You'd have to roll your own `groupby` for this, which I am not enthusiastic about. =)